[Bug] Unable to create a source for identifier 'Iceberg'.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

When the source is iceberg,sink is doris,the following error occurs.Unable to create a source for identifier 'Iceberg'.

### SeaTunnel Version

2.3.5

### SeaTunnel Config

```conf
env {
  job.mode = "BATCH"
  parallelism = 1
}
 
source {
  Iceberg {
    catalog_name = "ice"
    iceberg.catalog.config = {
        type = "hive"
        uri = "thrift://ip:port"
        warehouse = "hdfs://nameservice1/iceberg-warehouse/ice"
    }
    namespace = "tst"
    table = "aaa"
    iceberg.hadoop-conf-path = "/hadoop_conf"
  }
}
 
transform {
}
 
sink {
  Doris {
    sink.enable-2pc = "false"
    fenodes = "ip:port"
    username = "username"
    password = "password"
    table.identifier = "test.aaa"
    sink.label-prefix = "sink"
    doris.config {
      format = "json"
      read_json_by_line = "true"
   }
  }
}
```


### Running Command

```shell
${SEATUNNEL_HOME}/bin/seatunnel.sh --config xx.conf
```


### Error Exception

```log
2024-07-12 14:13:41,252 ERROR [o.a.s.c.s.SeaTunnel           ] [main] - Exception StackTrace:org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:202)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
    Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Iceberg'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:332)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:188)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:156)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:149)
        ... 2 more
    Caused by: java.lang.VerifyError: Stack map does not match the one at exception handler 20
    Exception Details:
      Location:
        org/apache/iceberg/hive/HiveCatalog.alterHiveDataBase(Lorg/apache/iceberg/catalog/Namespace;Lorg/apache/hadoop/hive/metastore/api/Database;)V @20: astore_3
      Reason:
        Type 'org/apache/hadoop/hive/metastore/api/NoSuchObjectException' (current frame, stack[0]) is not assignable to 'org/apache/thrift/TException' (stack map, stack[0])
      Current Frame:
        bci: @0
        flags: { }
        locals: { 'org/apache/iceberg/hive/HiveCatalog', 'org/apache/iceberg/catalog/Namespace', 'org/apache/hadoop/hive/metastore/api/Database' }
        stack: { 'org/apache/hadoop/hive/metastore/api/NoSuchObjectException' }
      Stackmap Frame:
        bci: @20
        flags: { }
        locals: { 'org/apache/iceberg/hive/HiveCatalog', 'org/apache/iceberg/catalog/Namespace', 'org/apache/hadoop/hive/metastore/api/Database' }
        stack: { 'org/apache/thrift/TException' }
      Bytecode:
        0x0000000: 2ab4 0023 2b2c ba00 9700 00b9 0029 0200
        0x0000010: 57a7 0061 4ebb 0037 592d 1238 04bd 0039
        0x0000020: 5903 2b53 b700 98bf 4ebb 003c 59bb 003d
        0x0000030: 59b7 003e 1299 b600 402b b600 4112 7ab6
        0x0000040: 0040 b600 422d b700 43bf 4eb8 0045 b600
        0x0000050: 46bb 003c 59bb 003d 59b7 003e 129a b600
        0x0000060: 402b b600 4112 7ab6 0040 b600 422d b700
        0x0000070: 43bf b1                                
      Exception Handler Table:
        bci [0, 17] => handler: 20
        bci [0, 17] => handler: 20
        bci [0, 17] => handler: 40
        bci [0, 17] => handler: 74
      Stackmap Table:
        same_locals_1_stack_item_frame(@20,Object[#59])
        same_locals_1_stack_item_frame(@40,Object[#59])
        same_locals_1_stack_item_frame(@74,Object[#68])
        same_frame(@114)
   
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at org.apache.iceberg.common.DynConstructors$Builder.impl(DynConstructors.java:149)
        at org.apache.iceberg.CatalogUtil.loadCatalog(CatalogUtil.java:221)
        at org.apache.iceberg.CatalogUtil.buildIcebergCatalog(CatalogUtil.java:284)
        at org.apache.seatunnel.connectors.seatunnel.iceberg.IcebergCatalogLoader.loadCatalog(IcebergCatalogLoader.java:55)
        at org.apache.seatunnel.connectors.seatunnel.iceberg.catalog.IcebergCatalog.open(IcebergCatalog.java:79)
        at org.apache.seatunnel.connectors.seatunnel.iceberg.source.IcebergSourceFactory.createSource(IcebergSourceFactory.java:100)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:111)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)
        ... 7 more
     
    2024-07-12 14:13:41,252 ERROR [o.a.s.c.s.SeaTunnel           ] [main] -
    ===============================================================================
   
   
   
    Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:202)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
    Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Iceberg'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:332)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:188)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:156)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:149)
        ... 2 more
    Caused by: java.lang.VerifyError: Stack map does not match the one at exception handler 20
    Exception Details:
      Location:
        org/apache/iceberg/hive/HiveCatalog.alterHiveDataBase(Lorg/apache/iceberg/catalog/Namespace;Lorg/apache/hadoop/hive/metastore/api/Database;)V @20: astore_3
      Reason:
        Type 'org/apache/hadoop/hive/metastore/api/NoSuchObjectException' (current frame, stack[0]) is not assignable to 'org/apache/thrift/TException' (stack map, stack[0])
      Current Frame:
        bci: @0
        flags: { }
        locals: { 'org/apache/iceberg/hive/HiveCatalog', 'org/apache/iceberg/catalog/Namespace', 'org/apache/hadoop/hive/metastore/api/Database' }
        stack: { 'org/apache/hadoop/hive/metastore/api/NoSuchObjectException' }
      Stackmap Frame:
        bci: @20
        flags: { }
        locals: { 'org/apache/iceberg/hive/HiveCatalog', 'org/apache/iceberg/catalog/Namespace', 'org/apache/hadoop/hive/metastore/api/Database' }
        stack: { 'org/apache/thrift/TException' }
      Bytecode:
        0x0000000: 2ab4 0023 2b2c ba00 9700 00b9 0029 0200
        0x0000010: 57a7 0061 4ebb 0037 592d 1238 04bd 0039
        0x0000020: 5903 2b53 b700 98bf 4ebb 003c 59bb 003d
        0x0000030: 59b7 003e 1299 b600 402b b600 4112 7ab6
        0x0000040: 0040 b600 422d b700 43bf 4eb8 0045 b600
        0x0000050: 46bb 003c 59bb 003d 59b7 003e 129a b600
        0x0000060: 402b b600 4112 7ab6 0040 b600 422d b700
        0x0000070: 43bf b1                                
      Exception Handler Table:
        bci [0, 17] => handler: 20
        bci [0, 17] => handler: 20
        bci [0, 17] => handler: 40
        bci [0, 17] => handler: 74
      Stackmap Table:
        same_locals_1_stack_item_frame(@20,Object[#59])
        same_locals_1_stack_item_frame(@40,Object[#59])
        same_locals_1_stack_item_frame(@74,Object[#68])
        same_frame(@114)
   
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at org.apache.iceberg.common.DynConstructors$Builder.impl(DynConstructors.java:149)
        at org.apache.iceberg.CatalogUtil.loadCatalog(CatalogUtil.java:221)
        at org.apache.iceberg.CatalogUtil.buildIcebergCatalog(CatalogUtil.java:284)
        at org.apache.seatunnel.connectors.seatunnel.iceberg.IcebergCatalogLoader.loadCatalog(IcebergCatalogLoader.java:55)
        at org.apache.seatunnel.connectors.seatunnel.iceberg.catalog.IcebergCatalog.open(IcebergCatalog.java:79)
        at org.apache.seatunnel.connectors.seatunnel.iceberg.source.IcebergSourceFactory.createSource(IcebergSourceFactory.java:100)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:111)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)
        ... 7 more
```


### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-doris/pom.xml
+++ b/seatunnel-connectors-v2/connector-doris/pom.xml
@@ -75,6 +75,12 @@
             <artifactId>seatunnel-thrift-service</artifactId>
             <version>${project.version}</version>
             <classifier>optional</classifier>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.thrift</groupId>
+                    <artifactId>libthrift</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.apache.seatunnel</groupId>

